app/assets/javascripts/components/revisions/diff_viewer.jsx

```diff
@@ -115,11 +116,13 @@ const DiffViewer = React.createClass({
         dataType: 'jsonp',
         url: diffUrl,
         success: (data) => {
-          const revisionData = data.query.pages[this.props.revision.mw_page_id].revisions[0];
+          const revisionData = data.query.pages[this.props.revision.mw_page_id].revisions;
```

Comment:
This and the following setState values would be clearer if you separately set up `revisionData` (which corresponds to `this.prop.revision`) and `firstRevisionData`.